### PR TITLE
tree-wide: coccinelle fixes

### PR DESCRIPTION
**Pull request by mrc0mmand**
Yet another batch of Coccinelle fixes.

### CODE REVIEW

**Comment by keszybz** [comment]
LGTM.

File: src/libsystemd-network/sd-dhcp-client.c [hunk 1]
```diff
@@ -238,7 +238,7 @@ int sd_dhcp_client_set_callback(
 int sd_dhcp_client_set_request_broadcast(sd_dhcp_client *client, int broadcast) {
         assert_return(client, -EINVAL);
 
-        client->request_broadcast = !!broadcast;
+        client->request_broadcast = broadcast;
```
Comment (bluca) [inline src/libsystemd-network/sd-dhcp-client.c:241]: Are we sure about this one? I see it often used to convert other types to bool

Comment (mrc0mmand) [reply src/libsystemd-network/sd-dhcp-client.c:241]: The rationalization comes from 5d904a6aaaceae7fe2f11d6b848b0dd45e3fd1c4 so I guess it's the way to go?

Comment (bluca) [reply src/libsystemd-network/sd-dhcp-client.c:241]: thanks, makes sense

**Review by bluca — APPROVED** [review]
lgtm

**Pull request merged**